R/explorer.R

```diff
@@ -26,45 +49,58 @@ explorer <- function(dataList=c(), addEnv=T, demo=F) {
     )
   )
 
-  formatFileList<-function(fileList){
-    fileList_formatted = list()
-    i<-1
-    for(f in fileList){
-      df<-get(f)
-      if(class(df)=="data.frame"){
-        fileObj = list(
-          File=f,
-          Rows = dim(df)[1],
-          Columns = dim(df)[2],
+  formatFileList <- function(fileList){
+    fileList_formatted <- list()
+    if (is.list(fileList)){
+      for (i in seq_along(fileList)){
+        df <- fileList[[i]]
+        fileList_formatted[[i]] <- list(
+          File = namesList[i],
+          Rows = nrow(df),
+          Columns = ncol(df),
+          json = jsonlite::toJSON(df)
+        )
+      }      
+    } else if (is.vector(fileList)){
+      for (i in seq_along(fileList)){
+        df <- get(fileList[i])
+        fileList_formatted[[i]] <- list(
+          File = fileList[i],
+          Rows = nrow(df),
+          Columns = ncol(df),
           json = jsonlite::toJSON(df)
         )
-        fileList_formatted[[i]]<-fileObj
-        i<-i+1
       }
     }
     return(fileList_formatted)
   }
 
+  
   # (2) Prep an array of objects for the user-specified files  (if length(dataArray)>0)
     data_list_formatted <- formatFileList(dataList)
     if(length(data_list_formatted)>0){
       rSettings[["settings"]][["files"]] = c(rSettings[["settings"]][["files"]], data_list_formatted)
     }
 
-
   # (3) Prep an array of objects for the environment files (if addEnv=T)
   if(addEnv){
-    env_list <- ls(pos=1)[sapply(ls(pos=1), function(x) class(get(x))) == 'data.frame']
-    env_list_formatted <- formatFileList(env_list)
-    if(length(env_list_formatted)>0){
-      rSettings[["settings"]][["files"]] = c(rSettings[["settings"]][["files"]], env_list_formatted)
+    if(length(ls(pos=1))>0){
+      env_list <- ls(pos=1)[sapply(ls(pos=1), function(x) inherits(get(x), "data.frame"))]
+      env_list_formatted <- formatFileList(env_list)
+      if(length(env_list_formatted)>0){
+        rSettings[["settings"]][["files"]] = c(rSettings[["settings"]][["files"]], env_list_formatted)
+      } else {
+        warning("No datasets to add from working environment.")
```

Comment:
Maybe update this to say something like, "No datasets to add from working environment; continuing with other user specified data sets."

Reply:
done.